### ISSUE TITLE
Unable to get the list of values

### ISSUE THREAD

**Issue by shaikadil95** [ricardbejarano/graphql_exporter#5]
Hi, i am using Graphql exporter, I am unable to get all the responses into prometheus. example response from graphql:

{
  "data": {
    "data": [
      {
        "id": 1,
        "name": "test",
        "template": 1,
        "nodes": [
          {
            "templateNode": {
              "name": "EB",
              "equipmentType": {
                "name": "5216"
              }
            }
          },
          {
            "templateNode": {
              "name": "CT",
              "equipmentType": {
                "name": "CT"
              }
            }
          },
          {
            "templateNode": {
              "name": "RU",
              "equipmentType": {
                "name": "RRUS_11_B4"
              }
            }
          },
          {
            "templateNode": {
              "name": "RU-B7",
              "equipmentType": {
                "name": "RRUS_11_B7"
              }
            }
          },
          {
            "templateNode": {
              "name": "E500-1",
              "equipmentType": {
                "name": "E500"
              }
            }
          },
          {
            "templateNode": {
              "name": "UE-1",
              "equipmentType": {
                "name": "UE_QC_5605"
              }
            }
          }
        ]
      }
    ]
  }
}

When I use promql query "query_data_nodes_templateNode_name" the response is: 
query_data_nodes_templateNode_name{data="0", endpoint="http://127.0.0.1:8080/graphql", instance="127.0.0.1:9199", job="graphql-exporter", nodes="0", value="EB"}

As you can see in the response there is only EB returned but not CT and RU.

How to solve this issue? any suggestions?

**Comment by ricardbejarano** [comment]
Hi @shaikadil95!
Thanks for using `graphql_exporter`, and apologies for the inconvencience.

I am able to reproduce the problem. I've also identified the cause.
I now need to think the proper code fix to it.

Thanks for raising this report.
I will get back to you on the following days.

**Comment by ricardbejarano** [comment]
Apologies on the delay.
I will begin work on this shortly.

**Comment by jasonyates-thg** [comment]
@ricardbejarano i have the same issue as the original poster attempting to pull Cloudflare Magic Transit tunnel stats. Was there ever a fixed pushed for this?

**Comment by ricardbejarano** [comment]
Nope! I'll put some time in this weekend. Apologies!

**Comment by ricardbejarano** [comment]
Fixed as of v1.2.2!

**Comment by ricardbejarano** [comment]
For future reference: it was calling r.Register multiple times for the same name+labels combination, which is not possible. Additionally, this failed silently because it wasn't handling the error returned by r.Register, which it cannot do since setGaugeValue has no return values to put it.

I've added a very simple map to store and reuse gauges throughout the duration of the scrape.

This is the response after doing so, when mocking with @shaikadil95's JSON:
```
# HELP query_data_id 
# TYPE query_data_id gauge
query_data_id{data="0",endpoint="http://127.0.0.1:5000/"} 1
# HELP query_data_name 
# TYPE query_data_name gauge
query_data_name{data="0",endpoint="http://127.0.0.1:5000/",value="test"} 1
# HELP query_data_nodes_templateNode_equipmentType_name 
# TYPE query_data_nodes_templateNode_equipmentType_name gauge
query_data_nodes_templateNode_equipmentType_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="0",value="5216"} 5216
query_data_nodes_templateNode_equipmentType_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="1",value="CT"} 1
query_data_nodes_templateNode_equipmentType_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="2",value="RRUS_11_B4"} 1
query_data_nodes_templateNode_equipmentType_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="3",value="RRUS_11_B7"} 1
query_data_nodes_templateNode_equipmentType_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="4",value="E500"} 1
query_data_nodes_templateNode_equipmentType_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="5",value="UE_QC_5605"} 1
# HELP query_data_nodes_templateNode_name 
# TYPE query_data_nodes_templateNode_name gauge
query_data_nodes_templateNode_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="0",value="EB"} 1
query_data_nodes_templateNode_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="1",value="CT"} 1
query_data_nodes_templateNode_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="2",value="RU"} 1
query_data_nodes_templateNode_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="3",value="RU-B7"} 1
query_data_nodes_templateNode_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="4",value="E500-1"} 1
query_data_nodes_templateNode_name{data="0",endpoint="http://127.0.0.1:5000/",nodes="5",value="UE-1"} 1
# HELP query_data_template 
# TYPE query_data_template gauge
query_data_template{data="0",endpoint="http://127.0.0.1:5000/"} 1
```
